src/main/java/sdu/se9/tv2/management/system/persistence/PersistenceCredit.java

```diff
@@ -69,119 +74,79 @@ public Credit createCredit(int programID, int personID, String roleName) throws
-            Credit element = this.credits.get(i);
+            PreparedStatement stmt = PersistenceDatabaseHelper.getConnection().prepareStatement("SELECT * FROM Credit WHERE programID ='" + programID + "' AND roleName ='" + roleName + "'");
+            ResultSet rs = stmt.executeQuery();
 
-            if (element.getPersonID() == personID) {
-                result.add(element);
+            if (!rs.next()) {
+                // No match
+                return null;
             }
 
-            if (result.size() >= maxCount) {
-                break;
-            }
-        }
+            int id = rs.getInt("id");
+            int program = rs.getInt("programID");
+            int person = rs.getInt("personID");
+            String role = rs.getString("roleName");
 
-        return result;
+            return new Credit(id, program, person, role);
     }
 
-    private void read() {
-        JSONObject obj = null;
-
-        // Try and read the file
-        try {
-            obj = this.persistence.read();
-        } catch (ParseException err) {
-            err.printStackTrace();
-        }
-
-        if (obj != null) {
-            // The file exists and has correct formatting, get lastID and list of producers
-
-            // obj.get("lastID") returns Long
-            this.lastID = Math.toIntExact((Long)obj.get("lastID"));
-
-            JSONArray objList = (JSONArray)obj.get("list");
-
-            ArrayList<Credit> parsedList = new ArrayList<Credit>();
 
-            // Parse producer list
-            Iterator<JSONObject> iterator = objList.iterator();
-            while (iterator.hasNext()) {
-                // Get element of the list array
-                JSONObject element = iterator.next();
-                // Parse the JSONObject using Producer.parseJSON
-                parsedList.add(Credit.parseJSON(element));
-            }
-
-            // Set producer list
-            credits = parsedList;
-        }
+    public ArrayList<Credit> getCreditsByPerson (int personID) throws SQLException {
+        return this.getCreditsByPerson(personID, Integer.MAX_VALUE);
     }
 
-    private void write() {
-        // Create JSONObject to save
-        JSONObject obj = new JSONObject();
-
-        // JSONArray that contains the producers
-        JSONArray list = new JSONArray();
-
-        // Go through producer list and parse as JSON objects
-        for (int i = 0; i < this.credits.size(); i++) {
-            list.add(Credit.parseJSON(this.credits.get(i)));
-        }
-
-        // Add lastID to JSON object
-        obj.put("lastID", lastID);
-        // Add list to JSON object
-        obj.put("list", list);
+    public ArrayList<Credit> getCreditsByPerson (int personID, int maxCount) throws SQLException {
+        Connection connection = PersistenceDatabaseHelper.getConnection();
+
+            PreparedStatement stmt = connection.prepareStatement("SELECT * FROM Credit WHERE personID =" + personID);
```

Comment:
Brug i stedet LIMIT = ? og sæt det til at være maxCount. Hvis i ikke gør det så fortæller i databasen at den skal finde alle krediteringerne for personen, også i selv om man kun leder efter et specifikt antal.